packages/linux_can/lib/src/data_classes.dart

```diff
@@ -566,6 +588,87 @@ class CanExtendedRemoteFrame extends CanFrame implements CanExtendedFrame, CanRe
   String toString() => 'CanExtendedRemoteFrame(id: $id)';
 }
 
+class CanFdBaseFrame extends CanFrame implements CanBaseFrame, CanDataFrame, CanFdFrame {
+  const CanFdBaseFrame({required this.id, required this.data, required this.flags})
+      : assert(0 <= data.length &&
```

Comment:
just one super minor thing I just saw :)

```suggestion
      : assert((id & ~CAN_EFF_MASK == 0) && 0 <= data.length &&
```